File: pset6/caesar/caesar.py
# import modules
import sys

def function():
    if len(sys.argv) == 2 :
        k = int(sys.argv[1])
        plain = input("plaintext: ")
        cypher = ""
        for i in range(0,len(plain)):
            if plain[i].isalpha():
                if plain[i].isupper():
                    a = ord(plain[i]) - 65
                    b = 65 + ((a + k) % 26)
                    cypher += chr(b)
                else:
                    c = ord(plain[i]) - 97
                    d = 97 + ((c + k) % 26)
                    cypher += chr(d)
            else:
                cypher += plain[i]
        print("ciphertext: {}".format(cypher))
        sys.exit(0)
    else:
        print("Usage: python caesar k")
        sys.exit(1)

if __name__ == '__main__':
    function()

File: pset2/caesa/caesar.c
# include <cs50.h>
# include <ctype.h>
# include <stdio.h>
# include <stdlib.h>
# include <string.h>

int main(int argc, string argv[])
{
    if(argc == 2)
    {
        int k = atoi(argv[1]);
        string plain = get_string("plaintext: ");
        int n = strlen(plain);
        char cypher[n+1];

        for(int i = 0; i < n; i++)
        {
            if(isalpha(plain[i]))
            {
                if(isupper(plain[i]))
                {
                    int b = plain[i]- 65;
                    int p = 65 + ((b + k) % 26);
                    char c = p;
                    cypher[i] = c;
                }
                else
                {
                    int a = plain[i] - 97;
                    int z = 97 + ((a + k) % 26);
                    char d = z;
                    cypher[i] = d;
                }
            }
            else
            {
                cypher[i] = plain[i];
            }
        }
        cypher[n] = '\0';
        printf("ciphertext: %s\n",cypher);
        return 0;
    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}

